bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperSTTService.java

```diff
@@ -0,0 +1,662 @@
+                                        locale.getLanguage());
+                            }
+                            if (config.singleUtteranceMode) {
+                                logger.debug("single utterance mode, ending transcription");
+                                transcription = tempTranscription;
+                                break;
+                            } else {
+                                // start a new transcription segment
+                                transcription += tempTranscription;
+                                tempTranscription = "";
+                            }
+                        } else if (nSegments == 0 && config.singleUtteranceMode) {
+                            logger.debug("Single utterance mode and no results, ending transcription");
+                            break;
+                        } else if (nSegments > 1) {
+                            // non reachable
+                            logger.warn("Whisper should be configured in single segment mode {}", nSegments);
+                            break;
+                        }
+                        // reset state to start with next segment
+                        voiceDetected = false;
+                        silenceSamplesCounter = 0;
+                        audioSamplesOffset = 0;
+                        logger.debug("Partial transcription: {}", tempTranscription);
+                        logger.debug("Transcription: {}", transcription);
+                    }
+                } finally {
+                    if (releaseContext) {
+                        ctx.close();
+                    }
+                }
+                // emit result
+                if (!aborted.get()) {
+                    sttListener.sttEventReceived(new RecognitionStopEvent());
+                    logger.debug("Final transcription: '{}'", transcription);
+                    if (!transcription.isBlank()) {
+                        sttListener.sttEventReceived(new SpeechRecognitionEvent(transcription.trim(), 1));
+                    } else {
+                        emitSpeechRecognitionError(sttListener);
+                    }
+                }
+            } catch (IOException e) {
+                logger.warn("Error running speech to text: {}", e.getMessage());
+                if (config.errorMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("Error"));
+                } else {
+                    emitSpeechRecognitionError(sttListener);
+                }
+            } catch (UnsatisfiedLinkError e) {
+                logger.warn("Missing native dependency: {}", e.getMessage());
+                if (config.errorMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("Error"));
+                } else {
+                    emitSpeechRecognitionError(sttListener);
+                }
```

Comment:
Do you need to check that config.errorMessage is not blank or would it be ok to let emitSpeechRecognitionError doing it ?